pkg/cmd/build/build.go

```diff
@@ -271,6 +271,16 @@ func OptsFromBuildInfo(manifestName, svcName string, b *model.BuildInfo, o *type
 		}
 	}
 
+	if len(b.Secrets) > 0 {
+		if o.Secrets == nil {
```

Comment:
nit: this is not necessary as go handles `append` for nil slices (not the same for nil maps though)